product_extended_segmentation/model/template.py

```diff
@@ -247,10 +237,82 @@ def _factor(factor, product_efficiency, product_rounding):
             tmpl_obj.write(
                 cr, uid, [product_tmpl_id.id], {'standard_price': price},
                 context=context)
-        tmpl_obj.write(
-            cr, uid, [product_tmpl_id.id], sgmnt_dict, context=context)
+        return self.ensure_change_price_log_messages(cr, uid, vals, context)
+
+    @api.multi
+    def _evaluate_threshold(self, bottom_th, computed_th, diff):
+        """
+        Checks if bottom threshold is reached by computed threshold
+        """
+        precision_id = self.env['decimal.precision'].precision_get('Account')
+        return float_is_zero(diff, precision_id) or\
+            (self.standard_price and diff < 0 and
+             computed_th > bottom_th)
+
+    @staticmethod
+    def _compute_threshold(std_price, new_price):
+        """
+         Computes price variation used for threshold checking
+        """
+        return std_price and abs((new_price - std_price) * 100 / std_price)\
+            or 0.0
+
+    @api.multi
+    def get_bottom_price_threshold(self):
+        """
+        Returns bottom threshold giving priority to product's company, if not,
+        user's company is taken
+        """
+        user = self.env['res.users'].browse(self._uid)
+        threshold = self.company_id.\
+            std_price_neg_threshold
+        if not threshold:
+            threshold = user.company_id.std_price_neg_threshold
+        return threshold
+
+    @api.model
+    def ensure_change_price_log_messages(self, vals):
+        """
+        Write in products (unless log_only is True) and then log messages
+        if segments or products (or both) were updated
+        """
+        sgmnts_dict = vals.get('sgmnts_dict', False)
+        log_only = vals.get('log_only', False)
+        new_price = vals.get('new_price', False)
+        prod_ids = vals.get('product_ids', False)
+        if prod_ids._name == 'product.template':
+            prod_ids = prod_ids.product_variant_ids
```

Comment:
this evaluation is not comming in the context?

Why are you asking for the _name?

I was waiting for the self._context.get('active_model') comparison, here it will send always to all prod_ids as I understand.